src/drivers/Kconfig

```diff
@@ -1,13 +1,118 @@
 menu "Drivers"
 
-config DMIC
-	bool "DMIC driver"
+config CAVS_DMIC
+	bool "Intel cAVS DMIC driver"
 	depends on CAVS
 	default n
 	help
-	  Select for DMIC driver
+	  Select this to enable Intel cAVS DMIC driver. The DMIC driver provides
+	  as DAI the SoC direct attach digital microphones interface.
+	  .
+	  .
 
-endmenu
+if CAVS_DMIC
+
+choice
+	prompt "FIR decimation coefficients set"
+	default CAVS_DMIC_FIR_FULL
+
+config CAVS_DMIC_FIR_FULL
+	bool "Full set"
+	select CAVS_DMIC_FIR_DECIMATE_BY_2
+	select CAVS_DMIC_FIR_DECIMATE_BY_3
+	select CAVS_DMIC_FIR_DECIMATE_BY_4
+	select CAVS_DMIC_FIR_DECIMATE_BY_5
+	select CAVS_DMIC_FIR_DECIMATE_BY_6
+	select CAVS_DMIC_FIR_DECIMATE_BY_8
+	help
+	  This option adds to previous all currently defined FIR
+	  coefficients sets to support sample rates 8 - 96 kHz with
+	  several microphone clock rates. The tables increase the size
+	  of the driver so this option should not be used in minimal
+	  systems.
+
+config CAVS_DMIC_FIR_LOW_MEMORY
+	bool "Small set"
+	select CAVS_DMIC_FIR_DECIMATE_BY_2
+	select CAVS_DMIC_FIR_DECIMATE_BY_6
+	help
+	  This option is used to minimize driver footprint but
+	  preserve support for 48 kHz and 16 kHz sample rates
+	  at typical 2.4 MHz microphone clock rate. The option
+	  enables decimation factors 2 and 6.
+
+config CAVS_DMIC_FIR_CUSTOM
+	bool "Custom set"
+	help
+	  This option is used to select each supported decimation
+	  factor.
+	  .
```

Comment:
No need to add these dots just to satisfy checkpatch, it's just guide, shorter description may be ok

Reply:
OK, I'll make a new version. The warnings from them are shown only in post commit check so committing this was possible.